src/app/clusters/sample-mei-server/sample-mei-server.cpp

```diff
@@ -0,0 +1,225 @@
+}
+
+void emberAfSampleMeiClusterServerInitCallback(chip::EndpointId endpoint)
+{
+    ChipLogProgress(Zcl, "Creating Sample MEI cluster, Ep %d", endpoint);
+    SampleMeiServer::Instance().RegisterEndpoint(endpoint);
+}
+
+void MatterSampleMeiClusterServerShutdownCallback(chip::EndpointId endpoint)
+{
+    // There's currently no whole-cluster shutdown callback. That would trigger
+    // call to `Shutdown`. Thus ep-based shutdown calls `UnregisterEndpoint`
+    ChipLogProgress(Zcl, "Shutting down Sample MEI cluster, Ep %d", endpoint);
+    SampleMeiServer::Instance().UnregisterEndpoint(endpoint);
+}
+
+// *****************************************************************************
+// SampleMeiContent
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace SampleMei {
+
+SampleMeiContent::SampleMeiContent() : SampleMeiContent(kInvalidEndpointId) {}
+
+SampleMeiContent::SampleMeiContent(EndpointId aEndpoint)
+{
+    endpoint = aEndpoint;
+
+    // Attribute default values
+    flipflop = false;
+}
+
+// *****************************************************************************
+// SampleMeiServer
+
+void SampleMeiServer::InvokeCommand(HandlerContext & ctxt)
+{
+    auto endpoint      = ctxt.mRequestPath.mEndpointId;
+    auto endpointIndex = EndpointIndex(endpoint);
+    if (endpointIndex == std::numeric_limits<size_t>::max())
+    {
+        ctxt.mCommandHandler.AddStatus(ctxt.mRequestPath, Protocols::InteractionModel::Status::UnsupportedEndpoint);
+        return;
+    }
+
+    switch (ctxt.mRequestPath.mCommandId)
+    {
+    case Commands::Ping::Id:
+        HandleCommand<Commands::Ping::DecodableType>(ctxt, [endpoint](HandlerContext & ctx, const auto & req) {
+            ChipLogProgress(Zcl, "Ping Command on Ep %d", endpoint);
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Success);
+        });
+        return;
+    case Commands::AddArguments::Id:
+        HandleCommand<Commands::AddArguments::DecodableType>(ctxt, [endpoint](HandlerContext & ctx, const auto & req) {
+            ChipLogProgress(Zcl, "AddArgumentsCommand on Ep %d", endpoint);
+            if (req.arg1 > UINT8_MAX - req.arg2)
+            {
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand);
+                return;
+            }
+
+            AddArgumentsResponse::Type response;
+            response.returnValue = static_cast<uint8_t>(req.arg1 + req.arg2);
+            ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+        });
+        return;
+    }
+}
+
+CHIP_ERROR SampleMeiServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err     = CHIP_NO_ERROR;
+    auto endpoint      = aPath.mEndpointId;
+    auto endpointIndex = EndpointIndex(endpoint);
+    if (endpointIndex == std::numeric_limits<size_t>::max())
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
```

Comment:
Still not sure why the errors for commands and attributes are different.  In other words, is there a reason this is not `CHIP_IM_GLOBAL_STATUS(UnsupportedEndpoint)`?  Similar for write.

Reply:
changed, thanks